controlplane/internal/controller/imageregistry/image_registry.go

```diff
@@ -0,0 +1,199 @@
+package imageRegistry
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/pelletier/go-toml"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	registryConfKey                = "registries.conf"
+	registryCertKey                = "ca-bundle.crt"
+	imageRegistryName              = "additional-registry"
+	registryCertConfigMapName      = "additional-registry-certificate"
+	registryCertConfigMapNamespace = "openshift-config"
+	imageConfigMapName             = "additional-registry-config"
+	imageDigestMirrorSetKey        = "image-digest-mirror-set.json"
+	registryCertConfigMapKey       = "additional-registry-certificate.json"
+	imageConfigKey                 = "image-config.json"
+)
+
+// CreateConfig creates a ConfigMap containing the manifests to be created on the spoke cluster.
+// The manifests that are in the ConfigMap include a ImageDigestMirrorSet CR with the mirror registry information,
+// a ConfigMap containing the additional trusted certificates for the mirror registry, and an image.config.openshift.io
+// CR that references the ConfigMap with the additional trusted certificate.
+// Successful creation of the ConfigMap will return the name of the created ConfigMap to be added as a manifest reference
+func CreateConfig(ctx context.Context, client client.Client, registryRef *corev1.LocalObjectReference, namespace string) (string, error) {
+
+	imageRegistryConfigMap := &corev1.ConfigMap{}
+	if err := client.Get(ctx, types.NamespacedName{Name: registryRef.Name, Namespace: namespace}, imageRegistryConfigMap); err != nil {
+		return "", err
+	}
+
+	additionalRegistryConfigMap, err := createImageRegistryConfig(imageRegistryConfigMap, namespace)
+	if err != nil {
+		return "", err
+	}
+
+	if err := client.Create(ctx, additionalRegistryConfigMap); err != nil {
+		return "", err
+	}
+	return additionalRegistryConfigMap.Name, nil
+}
+
+func createImageRegistryConfig(imageRegistry *corev1.ConfigMap, namespace string) (*corev1.ConfigMap, error) {
+	imageConfig := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      imageConfigMapName,
+			Namespace: namespace,
+		},
+		Data: make(map[string]string),
+	}
+	registryConf, ok := imageRegistry.Data[registryConfKey]
```

Comment:
What if the key is not there?

Reply:
As of right now, it skips adding the registry (in case the user wants to just provide an additional certificate and not a registry), but do you think we should not allow that case? Or we should bubble up a warning?

Reply:
Can you please expand on this use case? Why would the user just add a certificate and not a registry?

Asking because we also have AdditionalCertificateBundle - wouldn't this be the proper way to just add certs?

Reply:
Ah yes, the proper way to add additional certificates is through that field. Thanks!! So here we could return an error if the registry is not found, what do you think?

Reply:
Thanks!! Modified here https://github.com/openshift-assisted/cluster-api-agent/pull/24